src/inference/src/os/win/win_system_conf.cpp

```diff
@@ -7,21 +7,139 @@
+                mask_len = list_len;
+
+            } else if (info->Relationship == RelationProcessorCore) {
+                MaskToList(info->Processor.GroupMask->Mask);
+
+                if (_proc_type_table[ALL_PROC] >= _processors) {
+                    break;
+                }
+
+                if (0 == list[0]) {
+                    base_proc = _proc_type_table[ALL_PROC];
+                }
+
+                if (2 == list_len) {
+                    _cpu_mapping_table[list[0] + base_proc][CPU_MAP_PROCESSOR_ID] = list[0] + base_proc;
+                    _cpu_mapping_table[list[1] + base_proc][CPU_MAP_PROCESSOR_ID] = list[1] + base_proc;
+
+                    _cpu_mapping_table[list[0] + base_proc][CPU_MAP_SOCKET_ID] = _sockets;
+                    _cpu_mapping_table[list[1] + base_proc][CPU_MAP_SOCKET_ID] = _sockets;
+
+                    _cpu_mapping_table[list[0] + base_proc][CPU_MAP_CORE_ID] = _cores;
+                    _cpu_mapping_table[list[1] + base_proc][CPU_MAP_CORE_ID] = _cores;
+
+                    _cpu_mapping_table[list[0] + base_proc][CPU_MAP_CORE_TYPE] = HYPER_THREADING_PROC;
+                    _cpu_mapping_table[list[1] + base_proc][CPU_MAP_CORE_TYPE] = MAIN_CORE_PROC;
+
+                    _cpu_mapping_table[list[0] + base_proc][CPU_MAP_GROUP_ID] = group;
+                    _cpu_mapping_table[list[1] + base_proc][CPU_MAP_GROUP_ID] = group;
+
+                    _proc_type_table[MAIN_CORE_PROC]++;
+                    _proc_type_table[HYPER_THREADING_PROC]++;
+                    group++;
+
+                } else {
+                    _cpu_mapping_table[list[0] + base_proc][CPU_MAP_PROCESSOR_ID] = list[0] + base_proc;
+                    _cpu_mapping_table[list[0] + base_proc][CPU_MAP_SOCKET_ID] = _sockets;
+                    _cpu_mapping_table[list[0] + base_proc][CPU_MAP_CORE_ID] = _cores;
+                }
+                _proc_type_table[ALL_PROC] += list_len;
+                _cores++;
+
+            } else if ((info->Relationship == RelationCache) && (info->Cache.Level == 2)) {
+                MaskToList(info->Cache.GroupMask.Mask);
+
+                if (4 == list_len) {
+                    for (int m = 0; m < list_len; m++) {
+                        _cpu_mapping_table[list[m] + base_proc][CPU_MAP_CORE_TYPE] = EFFICIENT_CORE_PROC;
+                        _cpu_mapping_table[list[m] + base_proc][CPU_MAP_GROUP_ID] = group;
+                        _proc_type_table[EFFICIENT_CORE_PROC]++;
+                    }
+                    group++;
+
+                } else if (1 == list_len) {
+                    _cpu_mapping_table[list[0] + base_proc][CPU_MAP_CORE_TYPE] = MAIN_CORE_PROC;
+                    _cpu_mapping_table[list[0] + base_proc][CPU_MAP_GROUP_ID] = group;
+                    _proc_type_table[MAIN_CORE_PROC]++;
+                    group++;
+                }
+            }
+        }
+        delete[] base_ptr;
+    }
 };
 static CPU cpu;
+
```

Comment:
how can we test that new method works properly? Would be good to have unit tests for it

Reply:
so far we will validate on real platform with real Windows API during E2E test. 

Created ticket 100600 to follow up unit test case for parser of new method.

Reply:
tests should be implemented in the same PR as main functionality. No follow up tickets

Reply:
Not sure if there is a better solution for testing Microsoft defined structure in open source codebase. Please let us know if there is any reference.